features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/SnapshotProducer.kt

```diff
@@ -97,4 +101,50 @@ internal class SnapshotProducer(
             parentMappingContext
         }
     }
+
+    private fun resolvePrivacyOverrides(view: View, mappingContext: MappingContext): MappingContext {
+        val imagePrivacy =
+            try {
+                val privacy = view.getTag(R.id.datadog_image_privacy) as? String
+                if (privacy == null) {
+                    mappingContext.imagePrivacy
+                } else {
+                    ImagePrivacy.valueOf(privacy)
+                }
+            } catch (e: IllegalArgumentException) {
+                logInvalidPrivacyLevelError(e)
+                mappingContext.imagePrivacy
+            }
+
+        val textAndInputPrivacy =
+            try {
+                val privacy = view.getTag(R.id.datadog_text_and_input_privacy) as? String
+                if (privacy == null) {
+                    mappingContext.textAndInputPrivacy
+                } else {
+                    TextAndInputPrivacy.valueOf(privacy)
+                }
+            } catch (e: IllegalArgumentException) {
+                logInvalidPrivacyLevelError(e)
+                mappingContext.textAndInputPrivacy
+            }
+
+        return mappingContext.copy(
+            imagePrivacy = imagePrivacy,
+            textAndInputPrivacy = textAndInputPrivacy
+        )
+    }
+
+    private fun logInvalidPrivacyLevelError(e: Exception) {
+        internalLogger.log(
+            InternalLogger.Level.ERROR,
+            listOf(InternalLogger.Target.USER, InternalLogger.Target.TELEMETRY),
```

Comment:
Can user take any action when they see this error?

Reply:
It's in order for them to be aware that they have passed an invalid value (this would probably happen  in the future by them writing the wrong value in the xml, though atm we don't provide this option)